apps/example/src/domain/invoice.ts

```diff
@@ -1,190 +1,207 @@
+        case "Finalized":
+          throw new Error("Invoice is finalized")
+        case "Raised":
+          if (state.state.payments.has(data.reference)) return []
+          return [{ type: "PaymentReceived", data }]
+      }
+    }
+
+  export const finalize = (state: State): Event[] => {
     switch (state.type) {
       case "Initial":
         throw new Error("Invoice not found")
       case "Finalized":
-        throw new Error("Invoice is finalized")
+        return []
       case "Raised":
-        if (state.state.payments.has(data.reference)) return []
-        return [{ type: "PaymentReceived", data }]
+        return [{ type: "InvoiceFinalized" }]
     }
   }
-
-export const finalize = (state: State): Event[] => {
-  switch (state.type) {
-    case "Initial":
-      throw new Error("Invoice not found")
-    case "Finalized":
-      return []
-    case "Raised":
-      return [{ type: "InvoiceFinalized" }]
-  }
 }
 
-// Queries
-export type Model = {
-  amount: number
-  payer_id: string
-  finalized: boolean
-}
+namespace Query {
+  export type Model = {
+    amount: number
+    payer_id: string
+    finalized: boolean
+  }
 
-export const summary = (state: State): Model | null => {
-  switch (state.type) {
-    case "Initial":
-      return null
-    case "Raised":
-    case "Finalized":
-      return {
-        amount: state.state.amount,
-        payer_id: PayerId.toString(state.state.payer_id),
-        finalized: state.type === "Finalized",
-      }
+  export const summary = (state: Fold.State): Model | null => {
+    switch (state.type) {
+      case "Initial":
+        return null
+      case "Raised":
+      case "Finalized":
+        return {
+          amount: state.state.amount,
+          payer_id: PayerId.toString(state.state.payer_id),
+          finalized: state.type === "Finalized",
+        }
+    }
   }
 }
 
 export class Service {
-  constructor(private readonly resolve: (invoiceId: InvoiceId) => Decider<Event, State>) {}
+  constructor(
+    private readonly resolve: (invoiceId: InvoiceId) => Decider<Events.Event, Fold.State>,
+  ) {}
 
-  raise(id: InvoiceId, data: InvoiceRaised) {
+  raise(id: InvoiceId, data: Events.InvoiceRaised) {
     const decider = this.resolve(id)
-    return decider.transact(raiseInvoice(data), LoadOption.AssumeEmpty)
+    return decider.transact(Decide.raiseInvoice(data), LoadOption.AssumeEmpty)
   }
 
-  recordPayment(id: InvoiceId, data: Payment) {
+  recordPayment(id: InvoiceId, data: Events.Payment) {
     const decider = this.resolve(id)
-    return decider.transact(recordPayment(data))
+    return decider.transact(Decide.recordPayment(data))
   }
 
   finalize(id: InvoiceId) {
     const decider = this.resolve(id)
-    return decider.transact(finalize)
+    return decider.transact(Decide.finalize)
   }
 
   readInvoice(id: InvoiceId) {
     const decider = this.resolve(id)
-    return decider.query(summary)
+    return decider.query(Query.summary)
   }
 
```

Comment:
create and resolvecategory and random things like defaulting / loading config are `Factory` things?
I guess it makes more sense if you have a enough aggregates...